### PR TITLE
Add unit test for cluster get and add steps in workflows

### PR DESCRIPTION
**Pull request by chiayi**
## Why are these changes needed?

This change adds unit tests for the `get.go` file for the kubectl plugin. Also add steps in workflow to run these tests. 

Screenshot 2024-07-22 at 3 33 33 PM

---- 

Since there were some minor changes to the command, adding the manual sanity run here

Screenshot 2024-07-24 at 10 31 26 AM

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: kubectl-plugin/cmd/kubectl-ray.go [hunk 1]
```diff
@@ -5,13 +5,15 @@ import (
 
 	cmd "github.com/ray-project/kuberay/kubectl-plugin/pkg/cmd"
 	flag "github.com/spf13/pflag"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 func main() {
 	flags := flag.NewFlagSet("kubectl-ray", flag.ExitOnError)
 	flag.CommandLine = flags
+	ioStreams := genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
```
Comment (kevin85421) [inline kubectl-plugin/cmd/kubectl-ray.go:14]: `genericclioptions.IOStreams` seems to have already been deprecated https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions#IOStreams.

Comment (chiayi) [reply kubectl-plugin/cmd/kubectl-ray.go]: Changed it to use `genericiooptions.IOStreams` instead

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		"1",
+		"1",
+		"24Gi",
+		"<unknown>",
+	)
+
+	tf.FakeDynamicClient = fakedynamic.NewSimpleDynamicClient(runtime.NewScheme(), raycluster)
+
+	err := fakeClusterGetOptions.Run(context.Background(), tf)
+	assert.True(t, err == nil)
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:190]:
```suggestion
	assert.Nil(t, err)
```

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gosuri/uitable"
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:10]: Is this library commonly used? The repository does not seem to be actively maintained https://github.com/gosuri/uitable.

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+func TestRayClusterGetComplete(t *testing.T) {
+	testStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+	fakeClusterGetOptions := NewClusterOptions(testStreams)
+	fakeArgs := []string{"Expected", "output"}
+
+	*fakeClusterGetOptions.configFlags.Namespace = ""
+	fakeClusterGetOptions.AllNamespaces = false
+
+	err := fakeClusterGetOptions.Complete(fakeArgs)
+	if err != nil {
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:30]: `assert.Nil`

File: kubectl-plugin/pkg/cmd/cluster/get.go [hunk 5]
```diff
@@ -72,19 +76,12 @@ func (options *ClusterOptions) Validate() error {
...
-	restConfig, err := options.configFlags.ToRESTConfig()
-	if err != nil {
-		return err
-	}
-
-	// Retrieves the HTTPClient with the provided Rest client
-	dynamicClient, err := dynamic.NewForConfig(restConfig)
+func (options *ClusterOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	// Retrieves the dynamic client with factory.
+	dynamicClient, err := factory.DynamicClient()
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get.go:84]: Out of curiosity, what's a "dynamic client"?

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+)
+
+func TestRayClusterGetComplete(t *testing.T) {
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:21]: Can you add some comments to this test? It is not very straightforward to me. I understand the test after I trace the code.

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+	}
+
+	assert.True(t, fakeClusterGetOptions.AllNamespaces)
+	assert.Equal(t, fakeClusterGetOptions.args, fakeArgs)
+}
+
+func TestRayClusterGetValidate(t *testing.T) {
+	testStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+
+	testNS, testContext, testBT, testImpersonate := "test-namespace", "test-contet", "test-bearer-token", "test-person"
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:41]:
```suggestion
	testNS, testContext, testBT, testImpersonate := "test-namespace", "test-context", "test-bearer-token", "test-person"
```

File: kubectl-plugin/pkg/cmd/cluster/get_test.go [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+	// Fake directory for kubeconfig
+	fakeDir, err := os.MkdirTemp("", "fake-config")
+	if err != nil {
```
Comment (kevin85421) [inline kubectl-plugin/pkg/cmd/cluster/get_test.go:45]: `assert.Nil`

File: kubectl-plugin/pkg/cmd/cluster/get_test.go (hunk 3 above)
Comment (chiayi) [reply kubectl-plugin/pkg/cmd/cluster/get_test.go:10]: I don't think its commonly used. I was looking into using [this](https://pkg.go.dev/k8s.io/cli-runtime/pkg/printers) instead, which should be more reliable and more commonly used.

Comment (chiayi) [reply kubectl-plugin/pkg/cmd/cluster/get_test.go:10]: @kevin85421  Should I make the change in this PR or make a separate PR?

File: kubectl-plugin/pkg/cmd/cluster/get.go (hunk 5 above)
Comment (chiayi) [reply kubectl-plugin/pkg/cmd/cluster/get.go:84]: It provides interaction with non-standard Kubernetes resources so like CRDs and potentially unknown resources.

Comment (kevin85421) [reply kubectl-plugin/pkg/cmd/cluster/get.go:84]: Thank you for the explanation!

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**